kubeflow/trainer/rhai/traininghub.py

```
@@ -73,29 +124,494 @@ def _build_install_snippet(
+        Tuple of (apply_fn, handler_class) for testing purposes
+    """
+    import glob
+    import http.server
+    import json
+    import os
+    import subprocess
+    import threading
+
+    # Training Hub file constants (embedded for self-contained extraction)
+    # fmt: off
+    SFT_METRICS_FILE_PATTERN = "training_params_and_metrics_global*.jsonl"  # noqa: N806, F841
```

Ruff (0.14.8)

Unused `noqa` directive (unused: `F841`; non-enabled: `N806`)

Remove unused `noqa` directive

(RUF100)